enterprise/cmd/frontend/internal/app/app.go

```diff
@@ -163,30 +151,97 @@ func newGitHubAppCloudSetupHandler(db database.DB, apiURL *url.URL, client githu
 			return
 		}
 
+		responseServerError := func(msg string, err error) {
+			w.WriteHeader(http.StatusInternalServerError)
+			_, _ = w.Write([]byte(msg))
+			log15.Error(msg, "error", err)
+		}
+
 		a := actor.FromContext(r.Context())
-		if !a.IsAuthenticated() {
+		if !a.IsAuthenticated() || !envvar.SourcegraphDotComMode() {
```

Comment:
Should this be instead `&&` ?
```suggestion
		if !a.IsAuthenticated() && !envvar.SourcegraphDotComMode() {
```

Reply:
No, so this was from when the DotComMode GH App and the site-level GH App were running in parallel. The page it redirects to, in DotComMode, is one for when an admin approves an install without being authenticated.

If we aren't in DotComMode, it always redirects to this page (hence the `||`).

It can be simplified, but I want to focus on it when we tackle it in the UI/UX PR